flang/lib/Optimizer/Transforms/DoConcurrentConversion.cpp

```diff
@@ -99,91 +75,379 @@ class DoConcurrentConversion : public mlir::OpConversionPattern<fir::DoLoopOp> {
                   "constant LB, UB, and step values.");
     }
 
-    mlir::omp::LoopNestClauseOps clauseOps;
-    clauseOps.loopLBVar.push_back(rewriter.clone(*lbOp)->getResult(0));
-    clauseOps.loopUBVar.push_back(rewriter.clone(*ubOp)->getResult(0));
-    clauseOps.loopStepVar.push_back(rewriter.clone(*stepOp)->getResult(0));
-    clauseOps.loopInclusiveAttr = rewriter.getUnitAttr();
-    // ==== TODO (1) End ====
-    auto wsloopOp = rewriter.create<mlir::omp::WsloopOp>(doLoop.getLoc());
-    rewriter.createBlock(&wsloopOp.getRegion());
-    rewriter.setInsertionPoint(
-        rewriter.create<mlir::omp::TerminatorOp>(wsloopOp.getLoc()));
+    llvm::SmallVector<mlir::Value> liveIns;
+    collectLoopLiveIns(doLoop, liveIns);
+    assert(!liveIns.empty());
 
-    auto loopNestOp =
-        rewriter.create<mlir::omp::LoopNestOp>(doLoop.getLoc(), clauseOps);
+    mlir::IRMapping mapper;
+    mlir::omp::TargetOp targetOp = nullptr;
```

Comment:
Nit: No need for explicitly initializing to `nullptr`, as it's the default.
```suggestion
    mlir::omp::TargetOp targetOp;
```